### PR TITLE
Qa/nept 804

### PR DESCRIPTION
**Pull request by Fefaine**

### CODE REVIEW

**Comment by verbruggenalex** [comment]
Hi @Fefaine @voidtek , can you add the exceptions to the list of the qa roadmap and check if this branch passes Jenkins?

File: Jenkinsfile [hunk 1]
```diff
@@ -105,7 +105,7 @@ void executeStages(String label) {
         }
 
         stage('Check & Test ' + label) {
-            sh './bin/phpcs'
+            sh './bin/phpcs -s'
```
Comment (verbruggenalex) [inline Jenkinsfile]: If you want to have a nice summary on the bottom you can also add other reports, for example:
`./bin/phpcs --report=full --report=source --report=summary -s`
Just a tip: https://pear.php.net/manual/en/package.php.php-codesniffer.reporting.php

File: build.properties.dist [hunk 2]
```diff
@@ -177,8 +177,14 @@ phpcs.extensions = php inc module install info test profile theme css js
 # The default configuration file to generate.
 phpcs.config = ${project.basedir}/phpcs.xml
 
-# The coding standard to use.
-phpcs.standard = ${project.basedir}/phpcs-ruleset.xml
+# The locations for installed standards, delimited by comma.
```
Comment (jfhovinne) [inline build.properties.dist]: Why delimited by comma(s), instead of semicolon(s) ?

File: build.properties.dist [hunk 3]
```diff
@@ -177,8 +177,14 @@ phpcs.extensions = php inc module install info test profile theme css js
...
+phpcs.installed.paths = ${project.basedir}/vendor/ec-europa/qa-automation/phpcs/SubStandards
```
Comment (jfhovinne) [inline build.properties.dist]: Maybe we need a variable to store the qa-automation path first.

File: build.properties.dist [hunk 4]
```diff
@@ -177,8 +177,14 @@ phpcs.extensions = php inc module install info test profile theme css js
...
+
+# The coding standards to use, delimited by semicolons.
+phpcs.standards = Platform;${project.basedir}/phpcs-ruleset.xml
+
+# Returns a 0 error code when only warnings are found if enabled. Ment for CI.
```
Comment (jfhovinne) [inline build.properties.dist]: Confusing. Please list possible values and indicate their meaning.
Also 'Ment (meant) for CI' would need to be rephrased IMO.

File: build.properties.dist [hunk 5]
```diff
@@ -177,8 +177,14 @@ phpcs.extensions = php inc module install info test profile theme css js
...
+phpcs.passwarnings = 0
```
Comment (jfhovinne) [inline build.properties.dist]: Maybe phpcs.ignore_warnings ?

File: src/Phing/PhpCodeSnifferConfigurationTask.php [hunk 6]
```diff
@@ -246,6 +284,17 @@ public function setGlobalConfig($globalConfig) {
   }
 
   /**
+   * Sets the installed_paths configuration..
```
Comment (jfhovinne) [inline src/Phing/PhpCodeSnifferConfigurationTask.php]: 1 dot instead of 2 (also below)

File: composer.json [hunk 7]
```diff
@@ -1,10 +1,10 @@
 {
   "require": {
     "php": ">=5.4.0",
-    "drupal/coder": "8.2.6",
     "drupal/phingdrushtask": "^1.1",
     "drupol/phingbehattask": "^1.0",
     "drush/drush": "8.0.5",
+    "ec-europa/qa-automation": "3.0.0",
```
Comment (voidtek) [inline composer.json]: should be "ec-europa/qa-automation": "^3.0.0",

**Comment by drupol** [comment]
We've agreed to use ^3.0

File: src/Phing/PhpCodeSnifferConfigurationTask.php (hunk 6 above)
Comment (voidtek) [inline src/Phing/PhpCodeSnifferConfigurationTask.php]: :+1:

File: composer.json (hunk 7 above)
Comment (voidtek) [inline composer.json]: :+1:

File: src/Phing/PhpCodeSnifferConfigurationTask.php [hunk 8]
```diff
@@ -93,13 +108,32 @@ public function main() {
...
+      if (substr($standard, -4) === '.xml') {
+        if (file_exists($standard)) {
+          $element = $document->createElement('rule');
+          $element->setAttribute('ref', $standard);
+          $root_element->appendChild($element);
+        }
+      }
+      else {
+        foreach ($installedPaths as $installedPath) {
+          $ruleset = $installedPath . "/" . $standard . "/ruleset.xml";
```
Comment (jfhovinne) [inline src/Phing/PhpCodeSnifferConfigurationTask.php]: Should be single quotes, but not a blocker for the merge.

File: build.properties.dist [hunk 9]
```diff
@@ -177,10 +177,19 @@ phpcs.extensions = php inc module install info test profile theme css js
...
+phpcs.qaautomation.path = ${project.basedir}/vendor/ec-europa/qa-automation
 
-# Paths to check, delimited by semicolons.
+# The locations for installed standards, delimited by spaces, commas or semicolons.
+phpcs.installed.paths = ${phpcs.qaautomation.path}/phpcs/SubStandards
+
+# The coding standards to use, delimited by spaces, commas or semicolons.
+phpcs.standards = Platform;${project.basedir}/phpcs-ruleset.xml
+
+# Returns a 0 error code when only warnings are found if enabled. Ment for CI.
```
Comment (jfhovinne) [inline build.properties.dist]: "Ment for CI" => "Meant for CI" or remove this please.

File: build.properties.dist [hunk 10]
```diff
@@ -177,10 +177,19 @@ phpcs.extensions = php inc module install info test profile theme css js
 # The default configuration file to generate.
 phpcs.config = ${project.basedir}/phpcs.xml
 
-# The coding standard to use.
-phpcs.standard = ${project.basedir}/phpcs-ruleset.xml
+# The default path of qa-automation.
+phpcs.qaautomation.path = ${project.basedir}/vendor/ec-europa/qa-automation
```
Comment (jfhovinne) [inline build.properties.dist]: I guess phpcs.qa_automation.path is more readable.

File: build.properties.dist (hunk 9 above)
Comment (jfhovinne) [inline build.properties.dist]: And you need to document possible setting values, see my previous comment.
For instance "0 is disabled, 1 is enabled."

File: src/Phing/PhpCodeSnifferConfigurationTask.php [hunk 11]
```diff
@@ -246,19 +273,34 @@ public function setGlobalConfig($globalConfig) {
...
+   * Sets the installed_paths configuration.
+   *
+   * @param string $installedPaths
+   *   The paths in which the standards are installed.
+   */
+  public function setInstalledPaths($installedPaths) {
+    $this->installedPaths = $installedPaths;
+  }
+
+
```
Comment (jfhovinne) [inline src/Phing/PhpCodeSnifferConfigurationTask.php]: One newline instead of 2.

File: src/Phing/PhpCodeSnifferConfigurationTask.php (hunk 8 above)
Comment (voidtek) [inline src/Phing/PhpCodeSnifferConfigurationTask.php]: good check, I change that.

File: src/Phing/PhpCodeSnifferConfigurationTask.php (hunk 11 above)
Comment (voidtek) [inline src/Phing/PhpCodeSnifferConfigurationTask.php]: Changed.

File: build.properties.dist (hunk 10 above)
Comment (voidtek) [inline build.properties.dist]: ok. I change.

File: build.properties.dist (hunk 9 above)
Comment (voidtek) [inline build.properties.dist]: It's done.

**Comment by dxvargas** [comment]
:+1

**Comment by verbruggenalex** [comment]
:1st_place_medal:  :)

**Comment by voidtek** [comment]
This improvement is merged into master and in the tag 2.4.10

**Pull request merged**